dart.yaml

```diff
@@ -0,0 +1,137 @@
+package:
+  name: dart
+  version: 3.5.3
+  epoch: 0
+  description: The Dart SDK, including the VM, JS and Wasm compilers, analysis, core libraries, and more.
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+environment:
+  contents:
+    packages:
+      - bash
+      - bind-tools
+      - busybox
+      - ca-certificates
+      - ca-certificates-bundle
+      - curl
+      - git
+      - python3
+      - xz
+
+pipeline:
+  - name: "clone the source code"
+    runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib/
+      mkdir -p ${{targets.destdir}}/runtime
+      git clone --depth=1 https://chromium.googlesource.com/chromium/tools/depot_tools.git /home/depot_tools
+      export PATH=$PATH:/home/depot_tools
+      mkdir -p /home/dart-sdk
+      cd /home/dart-sdk
+      fetch dart
+      cd sdk
+      git fetch --tags --all
+      git checkout tags/${{package.version}} -b ${{package.version}}
+      gclient sync
+
+  - if: ${{build.arch}} == "x86_64"
+    working-directory: /home/dart-sdk/sdk
+    runs: |
+      ./tools/build.py --mode release --arch x64 create_sdk
+      mv out/ReleaseX64/dart-sdk ${{targets.destdir}}/usr/lib/dart
+      cp /lib64/ld-linux-x86-64.so.2 ${{targets.destdir}}/runtime/
```

Comment:
why is this not a symlink too?

Reply:
> I'd rather have our ca-certificate package ship symlink (not copy)

following our conversation, I am now using symlinks everywhere.

Reply:
@kranurag7 don't care what upstream do. In the end we ship an image. Doing a copy like this, destroys provenance & sbom.

Do they do chroot syscall into `/runtime` ? are they distributing `/runtime` as a container? well, for us container start at `/`.

Reply:
It's not chroot but a `cp` https://github.com/dart-lang/dart-docker/blob/a8ada07907fbfa139635fe90bfb31a236b7c2f34/Dockerfile-debian.template#L36-L51 (and this copying is what is being used at runtime inside scratch/static images via multistage builds) 

> are they distributing /runtime as a container?

Not as a separate container but they have docs for using image on dockerhub which does this and I think people are using that as a best practice.
https://hub.docker.com/_/dart

but to the original review here, I am thinking whether we should revert to copy again for these shared libraries (`/lib/ld-linux-aarch64.so.1` and `/lib64/ld-linux-x86-64.so.2`) because these two are also missing from scratch and static image.
_(mostly thinking about cases where the final executables are getting copied along with shared libraries including the above two in static/scratch image)_

Reply:
@kranurag7 

What is the difference between using `FROM cgr.dev/chainguard/dart` rather than from static and copy /runtime?

Note that our images are so minimal, that most of the time, when we build a runtime image, it really is "static + runtime" and nothing else.

In this current packaging do you have anything out-side of runtime? Cause potentially, I want to build two images here - "dart" that can be used for development and "dart-runtime" which one can use for deployment like our "jre" images and like a replacement for "FROM static + cp /runtime". Because then this dart-runtime image can have correct sbom; not vendor in shared library dependencies; and be rebuilt/upgraded automatically as glibc is upgraded.

Can you please create subpackage `name: ${{package.name}}-runtime` ship the dart files that are runtime only inside there, so split /runtime which contains dart original files only, without any `cp`. For any required things add symlinks + runtime dependencies. Ensure that main dart package runtime depends on dart-runtime. When building images, we will need to build two images dart and dart-runtime. In the dockerfile example our distroless way of building things then would be use `dart` image as build, and then copy to `dart-runtime`.

Your dart-runtime image, in sbom and installed database must have libc listed.

Also looking at the structure of runtime package, it looks more like regular package split with just a compat symlink from "/runtime -> /".

Does any of the above make sense at all? Or should I attempt to show you a prototype as a PR to your fork?